app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/TransferProgressFragment.java

```diff
@@ -0,0 +1,54 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.os.Bundle;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.kiwix.kiwixmobile.R;
+
+import java.util.ArrayList;
+
+/**
+ * Part of the local file sharing module, this fragment is used to display the progress of the
+ * file transfer. It displays a list of files along with their current status (as defined in the
+ * {@link FileItem} class.
+ * */
+public class TransferProgressFragment extends Fragment {
+
+  private ArrayList<FileItem> fileItems;
+  private RecyclerView filesRecyclerView;
+  private FileListAdapter fileListAdapter;
+
+  public TransferProgressFragment() {
+    // Required empty public constructor
+  }
+
+  public TransferProgressFragment(ArrayList<FileItem> fileItems) {
```

Comment:
this is rarely the way to send items to a fragment. Fragment.setArguments() is the standard

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/fc814159e7f452845cc636ffd5e8ef6c91fb49f1)
Any particular reason why static factory methods are used for fragments? Is it similar to why we use field-injection (since we control the creation of fragments)?

Reply:
We unfortunately don't actually control the creation, or more accurately the re-creation, of fragments https://stackoverflow.com/questions/11397099/whats-the-point-of-setarguments/14930903.
And nowadays we tend to use ViewModels and make Fragments just a dumb view for rendering data